### ISSUE TITLE
GraphQL complexity error on large site

### ISSUE THREAD

**Issue by LHSnow** [sveltia/sveltia-cms#178]
I run a fairly large static site; Hugo renders close to 4000 html pages. As I was experimenting a possible migration from DecapCMS (or at least give the users an option), I ran into "unexpected error" when loading site data. Gitlab backend reports

```json
{
    "errors": [
        {
            "message": "Query has complexity of 3788, which exceeds max complexity of 300"
        }
    ]
}
```

Query being (likely from [gitlab.js#L422](https://github.com/sveltia/sveltia-cms/blob/main/src/lib/services/backends/gitlab.js#L422))

```
 query ($paths: [String!]!) { project(fullPath: "my/repo") { repository { blobs(ref: "master", paths: $paths, after: "") { nodes { size rawTextBlob } pageInfo { endCursor hasNextPage } } } } } 
```

This looks like the Query tries to load every single markdown file in the repository in one query, rather than what the comment says, paginated.

The query error then causes `TypeError: Cannot read properties of undefined (reading 'project'` in the following statement, crashing the entire app.

Update: Reducing the config to only a single collection (the one with fewest pages) also reduces the Query complexity, but as the `media_folder` (with 800+ images) is also taken into consideration the error still occurs.

**Comment by kyoshino** [comment]
Pagination should be working as confirmed in #136, but unlike GitHub, GitLab has [very strict limitations](https://docs.gitlab.com/ee/administration/instance_limits.html). Will figure out what I can do to workaround the complexity issue. The query itself is simple, only retrieving the `size` and `rawTextBlob` properties, but I guess the `paths` array is too long.

**Comment by kyoshino** [comment]
Shipped the fix with [v0.36.1](https://github.com/sveltia/sveltia-cms/releases/tag/v0.36.1). Still, it would take long to load 4000+ files though. I’ve added a progress bar so you can at least see how it’s going.

**Comment by kyoshino** [comment]
And I have just released [v0.36.2](https://github.com/sveltia/sveltia-cms/releases/tag/v0.36.2) since I forgot the max number or records/files per request is 100.
